src/wp-includes/class-wp-customize-nav-menus.php

```diff
@@ -1559,7 +1559,7 @@ public function export_preview_data() {
 		$exports = array(
 			'navMenuInstanceArgs' => $this->preview_nav_menu_instance_args,
 		);
-		printf( '<script>var _wpCustomizePreviewNavMenusExports = %s;</script>', wp_json_encode( $exports ) );
+		wp_print_inline_script_tag( sprintf( /** @lang JavaScript */ 'var _wpCustomizePreviewNavMenusExports = %s;', wp_json_encode( $exports ) ) );
```

Comment:
PhpStorm flags the sprintf placeholder as a syntax error:

> Expression expected

![image](https://github.com/WordPress/wordpress-develop/assets/134745/2073099a-2c37-4a7d-a7df-d8b17e0416a7)

Reply:
These are language injection comments. Supported by PhpStorm, at least: https://www.jetbrains.com/help/phpstorm/using-language-injections.html

Reply:
Unless this is a standard, I think we should remove those annotations. We shouldn't change code to support arbitrary IDEs, unless this is something multiple independent tools use.

Reply:
But I don't think it hurts, and I believe at least a majority of WP developers have used PhpStorm. Such annotations are also in the #4824 by @spacedmonkey.

Reply:
The benefit is that if the IDE supports such annotations, they get syntax highlighting and IDE intelligence features. This is important when not using `<script>` as otherwise the IDE doesn't know. And since the JS here isn't being run through ESLint or other static analysis checks, I think we should try to get as many helps as possible from the IDE to catch bugs at development time.

Reply:
I get the reason for it, but I'm not sold on the justification to support an arbitrary IDE here. What if other IDEs also have a proprietary means for that that looks different? Would we add 2+ comments to those lines to satisfy all the IDEs? I don't think it's right for WP to "pick winners", which I think it does give preference to one by adding annotations that only that specific IDE benefits from.

It would be different if this was a standard pattern, but if it's not I'd prefer to remove it. I didn't spot it in #4824 before, but it applies there as well.

Reply:
Right, but I don't see this as being supporting an arbitrary IDE. If it's the most popular IDE for WP core development, it seems worthwhile.

Reply:
Even if it's the most popular one, it's still arbitrary if there's no standard.
> If it's the most popular IDE for WP core development, it seems worthwhile.

Maybe that is true, but we don't know that.

Reply:
> Maybe that is true, but we don't know that.

Bard says it is 😄 https://g.co/bard/share/e9560a1c3ce5

Anyway, I'd like to get a second opinion.